src/CanvasRenderingContext2d.cc

```diff
@@ -1600,6 +1641,127 @@ NAN_SETTER(Context2d::SetQuality) {
   cairo_pattern_set_filter(cairo_get_source(context->context()), filter);
 }
 
+/*
+ * Get current transform.
+ */
+
+NAN_GETTER(Context2d::GetCurrentTransform) {
+  Context2d *context = Nan::ObjectWrap::Unwrap<Context2d>(info.This());
+  Isolate *iso = Isolate::GetCurrent();
+
+  Local<Float64Array> arr = Float64Array::New(ArrayBuffer::New(iso, 48), 0, 6);
+  context->_getMatrix(arr);
+
+  const int argc = 1;
+  Local<Value> argv[argc] = { arr };
+  Local<Object> instance = Nan::NewInstance(_DOMMatrix.Get(iso), argc, argv).ToLocalChecked();
+
+  info.GetReturnValue().Set(instance);
+}
+
+/*
+ * Set current transform.
+ */
+
+NAN_SETTER(Context2d::SetCurrentTransform) {
+  Context2d *context = Nan::ObjectWrap::Unwrap<Context2d>(info.This());
+  Isolate *iso = Isolate::GetCurrent();
+  Local<Context> ctx = Nan::GetCurrentContext();
+
+  // TODO: Check if value is an instance of DOMMatrix
```

Comment:
There's no `InstanceOf` property of `v8::Object` in Node v6.x

Reply:
Good catch.

> Checking if one v8::Value is an instance of another v8::Value doesn't seem trivial, unless I'm missing something. It can be done by constructing a new function using global.Function constructor that performs JS instanceof operator on the arguments and returns the result, but it is very slow.

From C++ you'd ideally use `FunctionTemplate::HasInstance`, but you can't get the `FunctionTemplate` of a JS function that is passed into C++. Since v6 is EOL in April I'd suggest doing the easy method I suggested above with a preprocessor directive limiting it to Node.js 8+